src/librbd/crypto/openssl/OpenSSLDataCryptor.h

```diff
@@ -0,0 +1,40 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CRYPTO_OPENSSL_OPENSSL_DATA_CRYPTOR_H
+#define CEPH_LIBRBD_CRYPTO_OPENSSL_OPENSSL_DATA_CRYPTOR_H
+
+#include "librbd/crypto/DataCryptor.h"
+#include <openssl/evp.h>
+
+namespace librbd {
+namespace crypto {
+namespace openssl {
+
+class OpenSSLDataCryptor : public DataCryptor<EVP_CIPHER_CTX> {
```

Comment:
Nit: since it's namespaced to `openssl`, for clarity you can drop the duplicate `OpenSSL` prefix:
```suggestion
class DataCryptor : public crypto::DataCryptor<EVP_CIPHER_CTX> {
```